app/assets/javascripts/discourse/app/lib/utilities.js

```diff
@@ -62,11 +62,32 @@ export function avatarUrl(template, size, { customGetURL } = {}) {
   if (!template) {
     return "";
   }
-  const rawSize = getRawSize(translateSize(size));
+  const rawSize = getRawAvatarSize(translateSize(size));
   const templatedPath = template.replace(/\{size\}/g, rawSize);
   return (customGetURL || getURLWithCDN)(templatedPath);
 }
 
+let allowedSizes = null;
+
+export function getRawAvatarSize(size) {
+  allowedSizes =
+    allowedSizes ||
+    helperContext()
```

Comment:
```suggestion
  allowedSizes ??=
    helperContext()
```
(`||=` would be technically equivalent, but I think the nullish coalescing operator better describes what we're doing here)